lib/src/util/dart_type_utilities.dart

```diff
@@ -40,6 +40,12 @@ class DartTypeUtilities {
         element.allSupertypes.any(predicate);
   }
 
+  static bool extendsClass(DartType type, String className, String library) =>
+      type != null && type.name == className &&
+          type.element.library.name == library ||
+          (type is InterfaceType &&
+              extendsClass(type.superclass, className, library));
```

Comment:
Admittedly, it's a strange corner case, but should we support using Error and Exception as mixins? For example:

```
class E extends Object with Exception {
  static throw() {
    throw new E();
  }
}
```